### PR TITLE
Support swap files in qemu or bare-metal mode

### PR DESCRIPTION
**Pull request by Googulator**
This can help alleviate memory pressure when bootstrapping with high core counts.

### CODE REVIEW

**Comment by stikonas** [comment]
Hmm, codewise looks good, but generally I think swapping during build is bad for performance. I usually find that reducing number of threads to avoid swapping is more helpful. Or did you observe different behaviour in live-bootstrap?

**Comment by Googulator** [comment]
In the case of live-bootstrap, when going from 4 to 8 threads, the slight performance hit from occasionally hitting swap in GCC13 (the only place where 8 threads would otherwise go OOM) is far outweighed by the speed gain in Guile, especially on SSDs or other Flash storage. Haven't tested on an HDD though, results might be different there.

Also, Linux 4.9.10 is AFAIK already new enough to use swap in conjunction with disk caching even when there's no direct memory pressure, to increase performance.

File: steps/improve/swap.sh [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+# SPDX-FileCopyrightText: 2024 Gábor Stefanik <netrolller.3d@gmail.com>
+#
+# SPDX-License-Identifier: GPL-3.0-or-later
+#
+# Set up swap
+
+. /steps/bootstrap.cfg
+. /steps/env
+
+if ! test -f /swap
```
Comment (fosslinux) [inline steps/improve/swap.sh:13]: nit, convention has become in mainstream Linux distributions to locate this at `/swapfile`. Although they are equally descriptive using the convention is probably better.

Comment (Googulator) [reply steps/improve/swap.sh]: Done.

**Review by fosslinux — APPROVED** [review]

**Pull request merged**